### PR TITLE
feat(API):  GetRuneHolders endpoint with sorting and pagination

### PR DESCRIPTION
**Pull request by dcorral**
Example usage:

```json
{
    "jsonrpc": "2.0",
    "method": "Runes.GetRuneHolders",
    "params": {
        "runeTicker": "THEBESTRUNEEVER",
        "pagination": {"limit": 50, "skip": 0, "sort": "asc/desc"}
    },
    "id": 0
}
```

### CODE REVIEW

File: src/Methods/Runes/GetRuneHolders.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  params: Schema({
+    runeTicker: string,
+    pagination: schema.pagination.optional(),
+    sort: string.optional(),
+  }),
+  handler: async ({ runeTicker, pagination = {}, sort }) => {
+    pagination.limit ??= 100;
+    sort ??= "asc";
+    const _sort = sort === "asc" ? -1 : 1;
+    console.log(_sort);
```
Comment (danadi7) [inline src/Methods/Runes/GetRuneHolders.ts:18]: Remove the `console.log()` 🤣

File: src/Methods/Runes/GetRuneHolders.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    if (etching?.valid) {
+      const cap = etching.terms?.cap ?? 0n;
+      const amount = etching.terms?.amount ?? 0n;
+      const premine = etching.premine ?? 0n;
+      totalSupply = BigInt(cap) * BigInt(amount) + BigInt(premine);
+    }
+
+    const filter = {
+      runeTicker,
+      spentTxid: { $exists: false },
```
Comment (danadi7) [inline src/Methods/Runes/GetRuneHolders.ts:31]: Do we need an additional DB index for this? Seems like it may be a table scan.

**Review by danadi7 — COMMENTED** [review]
How's the performance of this endpoint? Did you test a few queries on the testnet DB?

File: src/Methods/Runes/GetRuneHolders.ts (hunk 2 above)
Comment (dcorral) [reply src/Methods/Runes/GetRuneHolders.ts:31]: could be a good idea but we will force a reindex right?

File: src/Methods/Runes/GetRuneHolders.ts (hunk 1 above)
Comment (dcorral) [reply src/Methods/Runes/GetRuneHolders.ts:18]: why?

**Comment by dcorral** [comment]
Barely tested on signet. More tests to come tomorrow on testnet.

**Pull request merged**